src/pow.cpp

```diff
@@ -29,12 +29,15 @@ unsigned int GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHead
         if (pblock->GetBlockTime() > pindexLast->GetBlockTime() + params.nPowTargetSpacing*2){
             return nProofOfWorkLimit;
         }
+        // Well, actually, it wasn't ever implemented properly in LBRYcrd.
+        // It doesn't work without the missing "else" statement found in upstream bitcoin.
+        // And that statement doesn't work correctly with our DifficultyAdjustmentInterval == 1.
+        // Hence, we are killing it at block 1100000.
     }
 
     // Go back the full period unless it's the first retarget after genesis.
-    int blockstogoback = params.DifficultyAdjustmentInterval()-1;
-    if ((pindexLast->nHeight+1) != params.DifficultyAdjustmentInterval())
-        blockstogoback = params.DifficultyAdjustmentInterval();
+    int blockstogoback = params.DifficultyAdjustmentInterval();
+    blockstogoback = std::min(blockstogoback, pindexLast->nHeight);
 
```

Comment:
What's the point here?  I'd actually think that all of this could probably be removed since we re-target every block and our full period is constant.

Reply:
Yes, it's only effective the very first time its hit (on the block right after genesis). We could special-case that block, but that would be making an assumption that DifficultyAdjustmentInterval returns 1. It does return 1 with our current settings, but if we're going to go down that road we might want to eliminate the parameters. And after working on this, I think that adjusting difficulty every block is unwise. It's too noisy, too prone to simple CPU fluctuations.

Reply:
Unwise or not, that's how the network operates.  I don't think it's any wiser to fork for a modification here, especially since mainnet has been running more or less sanely (i.e. nothing is broken at the moment).  And it's even less sane to do different retarget periods for testnet than for mainnet.

Reply:
Yes, the cap on the block height should make testnet and mainnet the same again. And I agree that a fork to make us not readjust difficulty every block is likely not worth it.

Reply:
I left my `min` call in as I think it's much more clear than what was there before, but feel free to revisit this in the upstream merge.